openedx_lti_tool_plugin/views.py

```diff
@@ -327,11 +327,41 @@ def get(
             unit_key: Unit key string.
 
         Returns:
-            HTTP response with rendered LTI courseware view.
+            Template response with rendered LTI courseware view
+            or HTTP 403 response if user is not enrolled.
+
+        Raises:
+            Http404: Course is not found, the unit is not in course.
         """
+        if not self.is_user_enrolled(request.user, course_id):
+            return HttpResponseForbidden(_(f'{request.user} is not enrolled to {course_id}'))
+
+        course_outline = self.get_course_outline(request, course_id)
+
+        # Get all course units and current unit on course.
+        units = []
+        unit_obj = None
+
+        for chapter in course_outline.get('children', []):
+            for sequence in chapter.get('children', []):
+                for unit in sequence.get('children', []):
+                    unit_id = unit.get('id')
+                    units.append(unit_id)
+
+                    # Get current unit ID from unit.
+                    if unit_key == unit_id:
+                        unit_obj = unit
+
+        # Return HTTP 404 if unit is not found.
+        if not unit_obj:
+            raise Http404()
+
         return self.render_to_response(
             {
-                'xblock_url': reverse(f'{AppConfig.name}:lti-xblock', args=[unit_key]),
+                'course_id': course_id,
```

Comment:
@alexjmpb what do you think about obtaining the course it from course_outline.id on the template instead of adding course_id to the template context?